pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableBuilder.java

```diff
@@ -91,9 +107,17 @@
   private ByteBuffer _currentRowDataByteBuffer;
 
   public DataTableBuilder(DataSchema dataSchema) {
+    CURRENT_VERSION = VERSION_3;
```

Comment:
I think we should make this configurable for until we migrate to v3 and remove v2 code. Default should be V3 (so that if anyone upgrades broker first. they are fine)

Reply:
Yes, it's configurable now. There is another constructor `public DataTableBuilder(DataSchema dataSchema, int version) {}`

Reply:
Now DataTableBuilder has a static function `setCurrentDataTableVersion()`, server can be configured to send either V2/V3 to datatable to broker.

Reply:
This doesn't look clean. CURRENT_VERSION should not be assigned in the constructor. It is a static final set to the current version # (3 for this PR).

Have a variable **`private static int _version = VERSION_3`** and assign that in the function  **`setCurrentDataTableVersion(int version)`** so that it can be overridden with the configured value.

Now in the builder, you check if _version is VERSION_2, call DataTableImplV2 else if _version is VERSION_3, call DataTableImplV3

This way you also remove CURRENT_VERSION and just keep VERSION_2, VERSION_3 and _version